osu.Framework/Testing/MenuTestScene.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.UserInterface;
+using osuTK.Input;
+
+namespace osu.Framework.Testing
+{
+    /// <summary>
+    /// A test scene that provides a set of helper functions and structures for testing a <see cref="Menu"/>.
+    /// </summary>
+    public abstract class MenuTestScene : ManualInputManagerTestScene
+    {
+        protected MenuStructure Menus;
+
+        public override IReadOnlyList<Type> RequiredTypes => new[]
+        {
+            typeof(Menu),
+            typeof(BasicMenu)
```

Comment:
```suggestion
            typeof(BasicMenu),
            typeof(MenuTestScene)
```
Class should probably declare self here to allow dynamic recompilation in derived tests.